src/config.py

```
@@ -0,0 +1,63 @@
+    Raises:
+        ConfigurationError: If configuration is invalid.
+    """
+    # Basic validation rules
+    if not config['api_base_url']:
+        raise ConfigurationError("Invalid or missing API base URL")
+
+    if config['request_timeout'] <= 0:
+        raise ConfigurationError("Request timeout must be a positive integer")
+
+    if config['max_retries'] < 0:
+        raise ConfigurationError("Max retries cannot be negative")
```

Pylint (3.3.7)

[convention] Final newline missing

(C0304)